src/auth/maptiler_hmacsha256/core/qgsauthmaptilerhmacsha256method.cpp

```diff
@@ -0,0 +1,188 @@
+{
+  return AUTH_METHOD_DESCRIPTION;
+}
+
+QString QgsAuthMapTilerHmacSha256Method::displayDescription() const
+{
+  return AUTH_METHOD_DISPLAY_DESCRIPTION;
+}
+
+bool QgsAuthMapTilerHmacSha256Method::updateNetworkRequest( QNetworkRequest &request, const QString &authcfg,
+    const QString &dataprovider )
+{
+  Q_UNUSED( dataprovider )
+  const QgsAuthMethodConfig mconfig = getMethodConfig( authcfg );
+  if ( !mconfig.isValid() )
+  {
+    QgsDebugMsg( QStringLiteral( "Update request config FAILED for authcfg: %1: config invalid" ).arg( authcfg ) );
+    return false;
+  }
+
+  const QString token = mconfig.config( QStringLiteral( "token" ) );
+  const QStringList splitToken = token.split( '_' );
+
+  if ( splitToken.count() != 2 )
+  {
+    QgsDebugMsg( QStringLiteral( "Update request config FAILED for authcfg: %1: config invalid" ).arg( authcfg ) );
+    return false;
+  }
+
+  const QString key = splitToken.at( 0 );
+  const QString secret = splitToken.at( 1 );
+
+  QUrl url = request.url();
+  QUrlQuery query( url.query() );
+  query.removeQueryItem( QStringLiteral( "key" ) );
+
+  QList<QPair<QString, QString> > queryItems = query.queryItems();
+
+  queryItems.append( {QStringLiteral( "key" ), key} );
+
+  query.setQueryItems( queryItems );
+  url.setQuery( query );
+
+  QString signature = calculateSignature( secret, url.url() );
+  request.setUrl( QString( url.url() + QStringLiteral( "&signature=" ) + signature ) );
+
+  return true;
+}
+
+void QgsAuthMapTilerHmacSha256Method::clearCachedConfig( const QString &authcfg )
+{
+  removeMethodConfig( authcfg );
+}
+
+void QgsAuthMapTilerHmacSha256Method::updateMethodConfig( QgsAuthMethodConfig &mconfig )
+{
+  if ( mconfig.hasConfig( QStringLiteral( "oldconfigstyle" ) ) )
+  {
+    QgsDebugMsg( QStringLiteral( "Updating old style auth method config" ) );
+  }
+
+  // NOTE: add updates as method version() increases due to config storage changes
+}
+
+QgsAuthMethodConfig QgsAuthMapTilerHmacSha256Method::getMethodConfig( const QString &authcfg, bool fullconfig )
+{
+  const QMutexLocker locker( &mMutex );
+  QgsAuthMethodConfig mconfig;
+
+  // check if it is cached
+  if ( sAuthConfigCache.contains( authcfg ) )
+  {
+    mconfig = sAuthConfigCache.value( authcfg );
+    QgsDebugMsg( QStringLiteral( "Retrieved config for authcfg: %1" ).arg( authcfg ) );
+    return mconfig;
+  }
+
+  // else build basic bundle
+  if ( !QgsApplication::authManager()->loadAuthenticationConfig( authcfg, mconfig, fullconfig ) )
+  {
+    QgsDebugMsg( QStringLiteral( "Retrieve config FAILED for authcfg: %1" ).arg( authcfg ) );
+    return QgsAuthMethodConfig();
+  }
+
+  // cache bundle
+  putMethodConfig( authcfg, mconfig );
+
+  return mconfig;
+}
+
+void QgsAuthMapTilerHmacSha256Method::putMethodConfig( const QString &authcfg, const QgsAuthMethodConfig &mconfig )
+{
+  const QMutexLocker locker( &mMutex );
+  QgsDebugMsg( QStringLiteral( "Putting token config for authcfg: %1" ).arg( authcfg ) );
```

Comment:
```suggestion
  QgsDebugMsgLevel( QStringLiteral( "Putting token config for authcfg: %1" ).arg( authcfg ), 2);
```

Also elsewhere, for all non-error debug messages

Reply:
done, also done in esri token auth where this code come from.